services/searcher.ts

```diff
@@ -213,6 +214,60 @@ class Searcher {
     ];
   }
 
+  /**
+   * Given a string, creates a list of queries that are either phrase queries or
+   * most_field queries. Phrases are between quotes, and use phrase matching.
+   * Most_field queries are everything else, and search the terms on all fields.
+   * else.
+   * @param query the string that is parsed into a list
+   * @returns an object containing a list of phrase_queries, and a field query.
+   */
+  parseQuery(query: string): ParsedQuery {
+    let matches = query.match(/"(.*?)"/gi);
+    //if there are no phrases, then make matches an empty list instead of being null
+    if (matches === null) {
+      matches = [];
+    }
+
+    const re = new RegExp(matches.join("|"), "gi");
+
+    //make sure theres no extra white space after removing the phrases.
+    const non_matches = query
+      .replace(re, "")
+      .replace('"', "")
+      .replace(/\s+/g, " ")
+      .trim();
+    const phraseQueries = [];
+
+    // go through the phrases, and make phrase queries with them.
+    for (let i = 0; i < matches.length; ++i) {
+      phraseQueries.push({
+        multi_match: {
+          query: matches[i].substring(1, matches[i].length - 1),
+          type: "phrase",
+          fields: this.getFields(),
+        },
+      });
+    }
```

Comment:
```suggestion
    const phraseQueries = matches.map((match) => {
        return {
          multi_match: {
            query: match[1],
            type: "phrase",
            fields: this.getFields(),
          },
        }
    });
```

The way `matchAll` works: for each match, it returns an array of matches. 

Each match contains, in order:
- The full match
- Any capturing groups (which on our case, is everything but the quotes)


Make sure to test this, since I'm just writing this code in github and haven't 100% checked it